com.unity.multiplayer.mlapi/Runtime/Messaging/InternalCommandContext.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using MLAPI.Serialization.Pooled;
+using System.Linq;
+
+namespace MLAPI.Messaging
+{
+    /// <summary>
+    /// A context used for building an internal command.
+    /// </summary>
+    internal struct InternalCommandContext : IDisposable
+    {
+        public PooledNetworkWriter NetworkWriter;
+
+        private ulong[] m_ClientIds;
+        private NetworkUpdateStage m_UpdateStage;
+        private MessageQueueContainer m_Owner;
+
+        public InternalCommandContext(PooledNetworkWriter writer, ulong[] clientIds, NetworkUpdateStage updateStage, MessageQueueContainer owner)
+        {
+            NetworkWriter = writer;
+            m_ClientIds = clientIds;
+            m_UpdateStage = updateStage;
+            m_Owner = owner;
+        }
+
+        public void Dispose()
+        {
+            Finalize();
+        }
+
+        public void Finalize()
+        {
+            if (m_Owner.NetworkManager.IsHost)
+            {
+                var containsServerClientId = m_ClientIds.Contains(m_Owner.NetworkManager.ServerClientId);
+                if (containsServerClientId && m_ClientIds.Length == 1)
+                {
+                    m_Owner.EndAddQueueItemToFrame(NetworkWriter, MessageQueueHistoryFrame.QueueFrameType.Inbound, m_UpdateStage);
```

Comment:
@becksebenius-unity @bendoyon We may be able to leverage this if `EndAddQueueItemToFrame` returns the message size to track a more granular level of bytes sent.